cardano-api/src/Cardano/Api/Eras.hs

```diff
@@ -0,0 +1,97 @@
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE TypeFamilies #-}
+
+-- The Shelley ledger uses promoted data kinds which we have to use, but we do
+-- not export any from this API. We also use them unticked as nature intended.
+{-# LANGUAGE DataKinds #-}
+{-# OPTIONS_GHC -Wno-unticked-promoted-constructors #-}
+
+module Cardano.Api.Eras
+  ( -- * Eras
+    Byron
+  , Shelley
+  , Allegra
+  , Mary
+  , AsType (AsPaymentKey, AsHash)
+
+    -- ** Hashes
+    -- | In Cardano most keys are identified by their hash, and hashes are
+    -- used in many other places.
+  , Hash(PaymentKeyHash)
+
+  , PaymentKey
+  , PaymentExtendedKey
+  , GenesisKey
+  , GenesisUTxOKey
+  , GenesisDelegateKey
+  , StakeKey
+  , StakePoolKey
```

Comment:
I know this is a draft, but these exports don't seem all that relevant to an `Eras` module. They seem like they'd fit better as part of a `Key` module or something.

Reply:
Agree, this is temporary though. I will likely do as you suggest and break off a `Key` module or something similar.